### PR TITLE
iio_demo: Fix xilinx sdk project build

### PR DESCRIPTION
**Pull request by mchindri**
Some compiler flags and libraries where not added to the project.
No it can build.

### CODE REVIEW

File: tools/scripts/generic.mk [hunk 1]
```diff
@@ -8,6 +8,10 @@ rwildcard = $(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2
 #                          COMMON INITIALIZATION
 #------------------------------------------------------------------------------
 
+#This compiler flags will be added to project if they are found in CFLAGS
+ADDITONAL_FLAGS = XILINX_PLATFORM ADUCM_PLATFORM ALTERA_PLATFORM \
```
Comment (buha) [inline tools/scripts/generic.mk:12]: I'm not sure if this is the right approach. If I want to add a completely new flag, I need to edit this line too which is not ideal.

We should propagate everything that starts with -D from CFLAGS to the SDK.

Comment (mchindri) [reply tools/scripts/generic.mk]: done

**Comment by mchindri** [comment]
v2:
 - Do not use ADDITONAL_FLAGS. Add all -D flags.
 - add commit to move commands inside if

**Comment by adrimbarean** [comment]
I think the first commit message needs to be updated to reflect the new approach.

**Comment by mchindri** [comment]
v3:
 - Update first commit message

**Review by adrimbarean — APPROVED** [review]
Looks good to me.

**Review by amiclaus — APPROVED** [review]

**Comment by buha** [comment]
These defines were added automatically by the scripts modified in this PR.
Screenshot from 2020-09-30 12-43-40

**Pull request merged**